pyomo/contrib/alternative_solutions/balas.py

```diff
@@ -0,0 +1,246 @@
+import pyomo.environ as pe
+from pyomo.common.collections import ComponentSet
+from pyomo.contrib.alternative_solutions import Solution
+import pyomo.contrib.alternative_solutions.aos_utils as aos_utils
+
+
+def enumerate_binary_solutions(
+    model,
+    *,
+    num_solutions=10,
+    variables="all",
+    rel_opt_gap=None,
+    abs_opt_gap=None,
+    search_mode="optimal",
+    solver="gurobi",
+    solver_options={},
+    tee=False,
+    quiet=True,
+    seed=None,
+):
+    """
+    Finds alternative optimal solutions for a binary problem using no-good
+    cuts.
+
+        Parameters
+        ----------
+        model : ConcreteModel
+            A concrete Pyomo model
+        num_solutions : int
+            The maximum number of solutions to generate.
+        variables: 'all' or a collection of Pyomo _GeneralVarData variables
+            The variables for which bounds will be generated. 'all' indicates
+            that all variables will be included. Alternatively, a collection of
+            _GenereralVarData variables can be provided.
+        rel_opt_gap : float or None
+            The relative optimality gap for the original objective for which
+            variable bounds will be found. None indicates that a relative gap
+            constraint will not be added to the model.
+        abs_opt_gap : float or None
+            The absolute optimality gap for the original objective for which
+            variable bounds will be found. None indicates that an absolute gap
+            constraint will not be added to the model.
+        search_mode : 'optimal', 'random', or 'hamming'
+            Indicates the mode that is used to generate alternative solutions.
+            The optimal mode finds the next best solution. The random mode
+            finds an alternative solution in the direction of a random ray. The
+            hamming mode iteratively finds solution that maximize the hamming
+            distance from previously discovered solutions.
+        solver : string
+            The solver to be used.
+        solver_options : dict
+            Solver option-value pairs to be passed to the solver.
+        tee : boolean
+            Boolean indicating that the solver output should be displayed.
+        quiet : boolean
+            Boolean indicating whether to suppress all output.
+        seed : int
+            Optional integer seed for the numpy random number generator
+
+        Returns
+        -------
+        solutions
+            A list of Solution objects.
+            [Solution]
+    """
+    if not quiet:  # pragma: no cover
+        print("STARTING NO-GOOD CUT ANALYSIS")
```

Comment:
See above.

Reply:
The easiest way to use the logger is to (at module scope):
```python
import logging
logger = logging.getLogger(__name__)
```
Then when you want to output text, instead of `print(msg)`, you would just do `logger.info(msg)` or `logger.debug(msg)`.  You can log at any integer level, but we tend to stick to the standard 5 (debug, info, warning, error, critical).

Reply:
Done